openvdb_cmd/vdb_tool/include/Parser.h

```diff
@@ -0,0 +1,1041 @@
+// Copyright Contributors to the OpenVDB Project
+// SPDX-License-Identifier: MPL-2.0
+
+////////////////////////////////////////////////////////////////////////////////
+///
+/// @author Ken Museth
+///
+/// @file Parser.h
+///
+/// @brief Defines various classes (Processor, Parser, Option, Action, Loop)
+///        for processing of command-line arguments.
+///
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef VDB_TOOL_PARSER_HAS_BEEN_INCLUDED
+#define VDB_TOOL_PARSER_HAS_BEEN_INCLUDED
+
+#include <iostream>
+#include <sstream>
+#include <string> // for std::string, std::stof and std::stoi
+#include <algorithm> // std::sort
+#include <random>
+#include <functional>
+#include <vector>
+#include <list>
+#include <set>
+#include <time.h>
+#include <initializer_list>
+#include <unordered_map>
+#include <iterator>// for std::advance
+#include <sys/stat.h>
+#include <stdio.h>
+
+#include <openvdb/openvdb.h>
+
+#include "Util.h"
+
+namespace openvdb {
+OPENVDB_USE_VERSION_NAMESPACE
+namespace OPENVDB_VERSION_NAME {
+namespace vdb_tool {
+
+// ==============================================================================================================
+
+/// @brief This class defines string attributes for options, i.e. arguments for actions
+struct Option {
+    std::string name, value, example, documentation;
+    void append(const std::string &v) {value = value.empty() ? v : value + "," + v;}
+};
+
+// ==============================================================================================================
+struct Action {
+    std::string            name;// primary name of action, eg "read"
+    std::string            alias;// alternate name for action, eg "i"
+    std::string            documentation;// documentation e.g. "read", "i", "files", "read files"
+    size_t                 anonymous;// index of the option to which the value of un-named option will be appended, e.g. files
+    std::vector<Option>    options;// e.g. {{"grids", "density,sphere"}, {"files", "path/base.ext"}}
+    std::function<void()>  init, run;// callback functions
```

Comment:
As a general comment, the member variables are usually included at the end of a struct definition.